### PR TITLE
- Return early in doSeek if docId found.

### PR DESCRIPTION
**Pull request by baldersheim**
- get_hits should not match anything before its current position.
- Cache bitvector size as a atomic does not optimize that well.

@havardpe PR

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp [hunk 1]
```diff
@@ -32,7 +32,7 @@ class DocumentWeightOrFilterSearchImpl : public DocumentWeightOrFilterSearch
     }
 
     std::unique_ptr<BitVector> get_hits(uint32_t begin_id) override {
-        return _children.get_hits(begin_id, getEndId());
+        return _children.get_hits(std::min(getEndId(), std::max(begin_id, getDocId())), getEndId());
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp:35]: this gives me bad memories

**Review by havardpe — COMMENTED** [review]
are we sure we want different sized bitvectors for the same docid range?

File: searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp:35]: It failed a conformance test without. I will check for alternative options.

Comment (baldersheim) [reply searchlib/src/vespa/searchlib/attribute/document_weight_or_filter_search.cpp]: Fixed

**Pull request merged**